### PR TITLE
Polish and rename ResolveIvyFactory

### PR DESCRIPTION
**Pull request by jvandort**
The ResolveIvyFactory is an older type that manages the building of ComponentResolver instances for external repositories. This type no longer is exclusive to ivy, and as such has been renamed to ExternalModuleComponentResolverFactory.

Further, we update this type to no longer require a ResolutionStrategyInternal, and instead pass in the relevant parameters from the resolution strategy. This decouples the factory from public/legacy types related to Configurations.

We also take this time to clean up artifact resolution queries a bit, which also uses this factory. We no longer create a detached configuration for each ARQ.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this

File: platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java [hunk 1]
```diff
@@ -188,7 +202,7 @@ public ResolverResults resolveBuildDependencies(ResolveContext resolveContext) {
         ResolvedLocalComponentsResultGraphVisitor localComponentsVisitor = new ResolvedLocalComponentsResultGraphVisitor(currentBuild, projectStateRegistry);
         DefaultResolvedArtifactsBuilder artifactsBuilder = new DefaultResolvedArtifactsBuilder(buildProjectDependencies);
 
-        ComponentResolvers resolvers = componentResolversFactory.create(resolveContext, ImmutableList.of(), consumerSchema);
+        ComponentResolvers resolvers = new ComponentResolversChain(getResolvers(resolveContext, Collections.emptyList()));
```
Comment (jvandort) [inline platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java:205]: `getResolvers` should wrap its result in `ComponentResolversChain` already

File: platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/query/DefaultArtifactResolutionQuery.java [hunk 2]
```diff
@@ -157,9 +146,17 @@ public ArtifactResolutionResult execute() {
...
             .collect(Collectors.toList());
 
-        ConfigurationInternal detachedConfiguration = configurationContainer.detachedConfiguration();
-        ResolutionStrategyInternal resolutionStrategy = detachedConfiguration.getResolutionStrategy();
-        ComponentResolvers componentResolvers = ivyFactory.create(resolutionStrategy, filteredRepositories, metadataHandler.getComponentMetadataProcessorFactory(), ImmutableAttributes.EMPTY, null, attributesFactory, componentMetadataSupplierRuleExecutor);
+        ResolutionStrategyInternal resolutionStrategy = resolutionStrategyFactory.create();
+        ComponentResolvers componentResolvers = externalResolverFactory.createResolvers(
+            filteredRepositories,
+            metadataHandler.getComponentMetadataProcessorFactory(),
+            resolutionStrategy.getComponentSelection(),
```
Comment (jvandort) [inline platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/query/DefaultArtifactResolutionQuery.java:153]: Add comment stating why we still use a resolution strategy here even though it is not user-configurable

File: platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java (hunk 1 above)
Comment (jvandort) [reply platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java:205]: Yes

**Review by octylFractal — APPROVED** [review]
LGTM as long as the two comments are resolved.

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Pull request merged**